src/main/java/pl/company/foodatu/meals/infrastructure/MealResponseKafkaProducer.java

```diff
@@ -0,0 +1,31 @@
+package pl.company.foodatu.meals.infrastructure;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.stereotype.Component;
+import pl.company.foodatu.meals.dto.MealResponse;
+
+@Component
+class MealResponseKafkaProducer implements MealPublisher {
+
+    private static final Logger logger = LoggerFactory.getLogger(MealResponseKafkaProducer.class);
+
+    private final KafkaTemplate<String, MealResponse> kafkaTemplate;
```

Comment:
We have to be careful with sending `MealResponse` directly to Kafka. Now, when changing `MealReponse` you have to remember that you may brake contract with kafka listeners. It's totally ok to have it and just be careful, but sometimes people introduce different types per each adapter, just to have more control and safety, e.g. `RestMealResponse`, `KafkaMealEvent` and so one

Reply:
I think, I will change this code to have KafkaMealEvent and RestMealResponse.